client/src/dashboard/RewritingRuleModal.js

```diff
@@ -177,9 +181,171 @@ const RewritingRuleModal = NiceModal.create(({user_id, rule=null, query=null}) =
       aceEditor.resize();
     });
   };
+
+  function findDistanceToSpaces(inputString, currentPosition) {
+    // Special case: if current character is a space
+    const isSpace = (inputString[currentPosition] === ' ');
+    const prevSpaceIndex = isSpace ? inputString.lastIndexOf(' ', currentPosition-1) : inputString.lastIndexOf(' ', currentPosition);
+    const nextSpaceIndex = isSpace ? inputString.indexOf(' ', currentPosition+1) : inputString.indexOf(' ', currentPosition);
+  
+    // Calculate the distance to the previous and next space characters
+    const distanceToPrevSpace = currentPosition - prevSpaceIndex;
+    const distanceToNextSpace = nextSpaceIndex - currentPosition;
   
+    return {
+      distanceToPrevSpace,
+      distanceToNextSpace,
+    };
+  };
+
+  const updateMarkers = () => {
+    // Clear previous markers
+    setq0Markers([]);
+    setq1Markers([]);
+
+    const q0Lines = (q0 === "") ? "" : q0.split('\n');
+    const q1Lines = (q1 === "") ? "" : q1.split('\n');
+    
+    if (! (q0Lines === "")) {
+      // Check on each code line's difference and update marker for each line
+      q0Lines.forEach((line, index) => {
+        const q1Line = (q1Lines === "" || q1Lines.length <= index) ? "" : q1Lines[index];
+        getDiffByLine(index, line, q1Line);
+      });
+    }
+  }
+
+  const getDiffByLine = (index, q0Line, q1Line) => {
+    // Get every character difference using diff library
```

Comment:
It would be nicer to add what the diffs look like with a simple example in the comment.